pkg/build/kube/bazelbuildbits.go

```diff
@@ -56,74 +60,114 @@ func (b *BazelBuildBits) Build() error {
 	// make sure we cd back when done
 	defer os.Chdir(cwd)
 
+	// TODO(bentheelder): we assume the host arch, but cross compiling should
+	// be possible now
+	bazelGoosGoarch := fmt.Sprintf("%s_%s", runtime.GOOS, runtime.GOARCH)
+
 	// build artifacts
 	cmd := exec.Command(
 		"bazel", "build",
-		// TODO(bentheelder): we assume linux amd64, but we could select
-		// this based on Arch etc. throughout, this flag supports GOOS/GOARCH
-		"--platforms=@io_bazel_rules_go//go/toolchain:linux_amd64",
+		fmt.Sprintf("--platforms=@io_bazel_rules_go//go/toolchain:%s", bazelGoosGoarch),
```

Comment:
Shouldn't this always be linux?

Reply:
just fixed that!

Reply:
`bazelGoosGoarch := fmt.Sprintf("linux_%s", runtime.GOARCH)` now 🙏 

this doesn't work on mac / windows anyhow though, FWIW. at least for the moment they need to not use `--type=bazel` when building kubernetes node images (we should document this in the docs!)